### ISSUE TITLE
How to show snippet description while using yas-insert-snippet?

### ISSUE THREAD

**Issue by kshenoy** [joaotavora/yasnippet#813]
`yas-insert-snippet` only shows snippet names which isn't very useful for me because if I'd known what I want, I'd just have typed it and be done with it. I tend to use yas-insert-snippet when I'm not sure of the name and in such cases having the description along with the name would be a lot more useful.

I know I can do the same thing with `yas-describe-tables`, search manually, then go back to my buffer and type it out. However, it'd be much easier and faster with yas-insert-snippet.

**Comment by npostavs** [comment]
I don't know which "description" you're talking about, `yas-describe-tables` shows abbrev key and snippet names. `yas-insert-snippet` prompts for snippet names.

There's a request to add a doc field to snippets, but it hasn't been done yet (#540).

**Comment by kshenoy** [comment]
Yes, I meant the snippet names. I guess #540 is going a step further (and maybe in a different, but welcome, direction). I'd love it if I could see the snippet names, which already exists today, while using `yas-insert-snippet`.

**Comment by npostavs** [comment]
Oh, it's a completing prompt, just hit `<tab>`.

**Comment by kshenoy** [comment]
@npostavs Oh, I got this totally wrong. It seems that `yas-insert-snippets` does show the names and not the keys. That's a little better. Is it possible to show the key as well?

**Comment by npostavs** [comment]
I think it might be possible, `(elisp) Programmed Completion` says something about an "annotation-function" that sounds promising.

Although I'm surprised that seeing a snippet key would ever give you more information than the snippet name.

**Comment by kshenoy** [comment]
> Although I'm surprised that seeing a snippet key would ever give you more information than the snippet name.

I  guess it's hard to explain but the best I can do is say that if I'm familiar with what a snippet does, what it's key is then I don't use `yas-insert-snippet` and I can directly type the key and be done with it.
However, if it's something I've created recently then seeing every bit of information helps. 

Additionally, broadly speaking there are two situations. First, in which I don't remember the key at all but I know conceptually the kind of snippet I want to insert eg. something to do with shared_ptr in C++. In this case, showing the name, like yas-insert-snippet does today, helps a lot since I can search for shared and pick one from the narrowed selection. However, since this doesn't show me the key, I still don't know what key to use for the next time onwards. I'll still have to pull up yas-describe-tables to figure that out. Had I both in front of me then it's easier to learn and pick this up.

The second situation is when I think I know the key (part or whole) but I'm not sure about it. In this case, I can run yas-insert-snippet, just type the key in and be shown either a smaller list from which I can make my selection.

I agree with you that just having the name should be sufficient but I thought this is something that'd be nice to have if it's not difficult.

**Comment by npostavs** [comment]
> since this doesn't show me the key, I still don't know what key to use for the next time onwards

Oh, good point, hadn't thought of that.

> this is something that'd be nice to have if it's not difficult.

I think it might not be that easy to implement (at least some research is needed, I can't immediately see how it should be done) and I don't know when I'll get around to it (patches welcome, of course).

> What does the `a` mean in yas-describe-tables

It seems to mean that the snippet has no `#condition` field. A `(y)` would be displayed for a fulfilled condition, and `(s)` for a non-fulfilled one.

https://github.com/joaotavora/yasnippet/blob/0.11.0/yasnippet.el#L2732-L2736

> and the auto-completion menu?

Which auto-completion menu?

**Comment by kshenoy** [comment]
> Which auto-completion menu?

[Here](http://imgur.com/JRJCumn)'s an example with autocomplete+yasnippet and the completion menu shows an 'a' in front of the snippets.

**Comment by npostavs** [comment]
Based on https://github.com/auto-complete/auto-complete/blob/v1.5.1/auto-complete-config.el#L170, I guess the autocomplete yasnippet backend arbitrarily puts an `a` for snippets.

**Comment by kshenoy** [comment]
@npostavs Then it's just a lucky coincidence that both are 'a'. Thanks for clarifying that. I was confused because I thought it was some kind of standard representation since two plugins were using it but I couldn't find anything about it.
